Talos 2 support ramstage+tests fixes

macpijan opened the pull request.

Commit:
util/cbfstool/flashmap/fmap.c: fix fmaptool endianness bugs on BE

This patch makes all accesses to the FMAP fields explicitly little endian.
It fixes issue where build on BE host produced different binary image than
on LE.

Signed-off-by: Marek Kasiewicz <marek.kasiewicz@3mdeb.com>
Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>
Change-Id: Ia88c0625cefa1e594ac1849271a71c3aacc8ce78

krystian-hebel committed:
arch/ppc64: initial development towards FIT payloads

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>
Change-Id: I91cdebf1135442a92a724eb864413dad8103fc92

krystian-hebel committed:
lib/fit_payload: comment out call to fit_update_memory()

Temporary comment out this call, it removes information about memory
associativity. Revert it later.

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

macpijan committed:
soc/power9: add missing SPDX license headers

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
soc/power9c: replace non-ASCII characters

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
soc/power9/chip.c: remove final empty line

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan commented on src/mainboard/raptor-cs/talos-2/board.fmd:
```diff
@@ -11,7 +11,7 @@
 FLASH@0 CONFIG_ROM_SIZE {
 	BIOS@0 CONFIG_ROM_SIZE {
-#if !CONFIG_BOOTBLOCK_IN_SEEPROM
+#if !CONFIG(BOOTBLOCK_IN_SEEPROM)
```
@krystian-hebel I guess i tried to make the linter happy and made the build fail at the same time: 
https://github.com/3mdeb/coreboot/runs/2719266434?check_suite_focus=true#step:4:327

macpijan replied:
I can see no other .fmd files in tree using any kind of CONFIG dynamically. And the linter checks for the syntax it expects to find in the regular source files

macpijan replied:
I can see that in case of different platforms rather a set of different .fmd files is provided and the proper one is used. Maybe we should do that as well?

krystian-hebel replied:
I we have to... Then in Kconfig we should add different defaults depending on `BOOTBLOCK_IN_SEEPROM` as well.

macpijan replied:
@krystian-hebel Please take a look if it makes sense: https://github.com/3mdeb/coreboot/pull/65/commits/8a2c5569333f608594d3c94f1d280608f5957364

macpijan replied:
It is this one after rebasing: https://github.com/3mdeb/coreboot/pull/65/commits/bccd98ed5b500338b3954331db7ba13e322f10e1

Both variants (bootblock in PNOR/EEPROM) are already passing in the CI job.

macpijan committed:
mainboard/talos-2: split board.fmd in two files

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
configs/config.raptor-cs-talos-2-bootblock-in-seeprom: add

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
.github/workflows/build.yml: add upstream tests and SEEPROM build

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
.github/workflows/build.yml: add FreeBSD job

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
.github/workflows/build.yml: build more tools in FreeBSD

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

krystian-hebel committed:
util/ifdtool/Makefile.inc: define _DEFAULT_SOURCE for compiling FMAP

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>
Change-Id: If26ddc3bdc394efd120967194b18cecf66c46355

macpijan committed:
Merge pull request #66 from 3mdeb/ifdtool_build_fix

util/ifdtool/Makefile.inc: define _DEFAULT_SOURCE for compiling FMAP

macpijan committed:
.github/workflows/build.yml: install pkgconf and bash for FreeBSD

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

krystian-hebel committed:
lib/fit_payload.c: move call to dt_apply_fixups()

This patch moves a call to dt_apply_fixups() further into FDT preparation.
It allows the fixups to modify 'chosen' and 'memory' nodes, which weren't
yet present when dt_apply_fixups() was called earlier.

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>
Change-Id: I13caa4e033bef07eaab33516d087de642e10e093

Commit:
Fit payload building (#67)

* Prepare files for building FIT payload

* payloads/external/skiboot: include choice in menu

* payloads/external/skiboot: Add building of uImage

* payloads/external/skiboot: Include skiboot in build

* Select correct compression algorithm

* Change ROM size to 2048 and fix skiboot cleaning

* Fixes after review part 1

* Add cleaning of fdt.bin

* Add board-specific options to Kconfig

* Build skiboot payload intermediate files inside build directory

* Fix skiboot clean

Co-authored-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel committed:
soc/ibm/power9: add required "ibm,chip-id" property to the device tree

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>
Change-Id: Ic45fae90350a968f0a1c813d673c139f58db21e3

krystian-hebel committed:
soc/ibm/power9: fill memory table with real values

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>
Change-Id: I52019bba54cb62134d9409c47363f7604d8e130a

krystian-hebel committed:
mainboard/talos-2: shorten DTS file name, remove memory nodes

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>
Change-Id: I0bdac7ca3125fe299c6f3fae56fcd9034b567f1c

krystian-hebel committed:
soc/power9/rom_media.c: remove obsolete comment

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>
Change-Id: I1d6e6c8467a83ec77a0da4e02a786e4ac08a28de

macpijan committed:
.github/workflows/build.yml: build jenkins tests for one mb only

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
docker/coreboot-sdk: install u-boot-tools

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
.github/workflows/build.yml: use 3mdeb/coreboot-sdk

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
.github/workflows/build.yml: fix 3mdeb/coreboot-sdk tag

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
mb/talos-2/devicetree.dts: Add SPDX identifier

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
payloads/external/skiboot/Makefile: clean skiboot only if it is present

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>
